Remotion/Validation/UnitTests/Rules/ObjectValidationRuleTest.cs

```diff
@@ -1,21 +1,176 @@
+          .Returns(new[] { failure1 });
+
+      var rule = new ObjectValidationRule<object>(null, new[] { validator1Stub.Object });
+
+      var result = rule.Validate(validationContext);
+      Assert.That(result, Is.EquivalentTo(new[] { failure1 }));
+    }
+
+    [Test]
+    public void Validate_WithNoValidators_ReturnsEmptyResult ()
+    {
+      var context = new ValidationContext(new object());
+      var rule = new ObjectValidationRule<object>(null, new List<IObjectValidator>());
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithNullAsInstanceToValidate_ReturnsEmptyResultAsValidatorsAreNotCalled ()
+    {
+      var context = new ValidationContext(null);
+
+      var validatorStub = new Mock<IObjectValidator>();
+      validatorStub
+          .Setup(_=>_.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new [] {new ObjectValidationFailure(
+                              15,
+                              "15 is not a number",
+                              "Localized validation message: 15 is not a number.")});
+
+      var rule = new ObjectValidationRule<object>(null, new [] {validatorStub.Object});
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithUnfulfilledCondition_ReturnsEmptyResultAsValidatorsAreNotCalled ()
+    {
+      Func<object, bool> condition = _ => false;
+      var context = new ValidationContext(new object());
+
+      var validatorStub = new Mock<IObjectValidator>();
+      validatorStub
+          .Setup(_=>_.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new [] {new ObjectValidationFailure(
+              15,
+              "15 is not a number",
+              "Localized validation message: 15 is not a number.")});
+
+      var rule = new ObjectValidationRule<object>(condition, new [] {validatorStub.Object});
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void IsActive_WithoutInstanceToValidate_ReturnsFalse ()
+    {
+      Func<object, bool> condition = _ => true;
+
+      var context = new ValidationContext(null);
+      var rule = new ObjectValidationRule<object>(condition, new List<IObjectValidator>());
+
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.False);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void IsActive ()
+    public void IsActive_WithNoCondition_ReturnsTrue ()
     {
-      // consider better name than "IsActive"
+      var context = new ValidationContext(new object());
+      var rule = new ObjectValidationRule<object>(null, new List<IObjectValidator>());
+
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.True);
+    }
+
+    [Test]
+    public void IsActive_ReturnsValueFromCondition ()
```

Comment:
This would be best in two tests, or using TestCase attribute or using BooleanObjectMother -> we try to avoid having two system under test isntances in the same test method.